library/core/src/mem/mod.rs

```diff
@@ -874,6 +874,38 @@ pub const fn replace<T>(dest: &mut T, src: T) -> T {
     }
 }
 
+/// Replaces the value in `dest` by the value returned by `f`.
+///
+/// `f` receives the current referenced value of `dest` as its argument.
+///
+/// * If you want to replace the values of two variables, see [`swap`].
+/// * If you want to replace with a default value, see [`take`].
+/// * If you want replace without the function, see [`replace`].
+///
+/// # Examples
+///
+/// A simple example:
+///
+/// ```
+/// use std::mem;
+///
+/// let mut v: Vec<i32> = vec![1, 2];
+///
+/// mem::replace(&mut v, |mut v| { v.push(3); v });
```

Comment:
```suggestion
/// mem::reshape(&mut v, |mut v| { v.push(3); v });
```
Should this be `reshape`?